### PR TITLE
switch to `argparse` for CLI

### PR DESCRIPTION
**Pull request by aniketmaurya**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

JSONARGPARSE for CLI adds a extra dependency. This PR switches to argparse to keep the library lean. 

### Before
image

### After
image

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]

File: tests/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    with open(dummy_server_file, "w") as f:
+        f.write("# Dummy server file for testing\n")
+
+    monkeypatch.setattr("sys.argv", ["litserve", "dockerize", dummy_server_file])
+    main()
+    captured = capsys.readouterr()
+    assert "Dockerfile created successfully" in captured.out, "CLI did not create Dockerfile"
+    assert os.path.exists("Dockerfile"), "CLI did not create Dockerfile"
+
+    os.remove(dummy_server_file)
```
[the remark this answers is not in the document]
Comment (aniketmaurya) [reply tests/test_cli.py:28]: yeah, right! let me update

File: src/litserve/__main__.py [hunk 2]
```diff
@@ -11,16 +11,50 @@
...
+    )
+    dockerize_parser.add_argument(
+        "--gpu",
+        default=False,
+        action="store_true",
+        help="Whether to use a GPU-enabled Docker image. Defaults to false.",
+    )
+    dockerize_parser.set_defaults(func=lambda args: dockerize(args.server_filename, args.port, args.gpu))
+
+    # Add "litserve build or litserve serve" sub-commands here
```
Comment (lantiga) [inline src/litserve/__main__.py:47]: remove comments?

**Review by lantiga — APPROVED** [review]
Looks good, remember to remove commented code

File: src/litserve/__main__.py (hunk 2 above)
Comment (aniketmaurya) [reply src/litserve/__main__.py:47]: ah, I thought it would be helpful for future contributors. Will drop it.

**Comment by Borda** [comment]
@aniketmaurya let's use [Fire](https://github.com/google/python-fire) as we already have it for SDK so it would be zero dependency

**Comment by aniketmaurya** [comment]
@Borda Yes, but if someone wants to dockerize their server and deploy it, the Docker image shouldn't include any external dependencies.

**Pull request merged**